### PR TITLE
feat(build): added pr linting

### PR DESCRIPTION
**Pull request by hasanaburayyan**
Adds PR lint to enforce good PR titles for release notes sake

### CODE REVIEW

File: .github/workflows/pr-lint.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            grammar
+            docs
+          types: |-
+            feat
+            fix
+            docs
+            chore
+            revert
+            build
+          subjectPattern: ^[^A-Z][^:]+[^.]$
```
Comment (Tirithel) [inline .github/workflows/pr-lint.yml]:
```suggestion
          subjectPattern: ^[^:]+$
```

I'm not sure I care about the capitalization nor the ending with a period. Thoughts?

Comment (hasanaburayyan) [reply .github/workflows/pr-lint.yml]: for me its consistency. When looking through release notes id prefer to see the conventions be consistent then you never have someone say "When should I use Feat vs feat"

Comment (Tirithel) [reply .github/workflows/pr-lint.yml]: This is for just the subject though. The scope and type are still controlled.

Comment (hasanaburayyan) [reply .github/workflows/pr-lint.yml]: Yea you are right. I still think there is a consistency thing here but lets let hell break loose 😁

**Review by Tirithel — APPROVED** [review]

**Pull request merged**